test/recording-history.js

```diff
@@ -0,0 +1,300 @@
+  t.context.paths = [existingPath];
+
+  recordingHistory.set('recordings', [{filePath: existingPath}, {filePath: missingPath}]);
+
+  t.deepEqual(getPastRecordings(), [{filePath: existingPath}]);
+  t.deepEqual(recordingHistory.get('recordings'), [{filePath: existingPath}]);
+});
+
+test('checkForActiveRecording with no recording', async t => {
+  t.false(await checkForActiveRecording());
+});
+
+test('checkForActiveRecording with playable recording', async t => {
+  const fakeService = {
+    title: 'Fake Service',
+    cleanUp: sinon.fake()
+  };
+
+  const fakePlugin = {
+    name: 'kap-fake-plugin',
+    recordServices: [fakeService]
+  };
+
+  plugins.getRecordingPlugins = sinon.fake.returns([fakePlugin]);
+
+  recordingHistory.set('activeRecording', {
+    filePath: incomplete,
+    name: 'Incomplete',
+    date: new Date().toISOString(),
+    apertureOptions: {},
+    plugins: {
+      'kap-fake-plugin': {
+        'Fake Service': {
+          some: 'state'
+        }
+      }
+    }
+  });
+
+  const checkPromise = checkForActiveRecording();
+
+  const dialogState = await dialog.waitForDialog();
+  t.true(dialogState.detail.includes('playable'));
+
+  // Don't delete it until user is done interacting with the dialog
+  t.true(recordingHistory.has('activeRecording'));
+
+  await dialog.fakeAction(dialogState.buttons.findIndex(b => b.label && b.label.toLowerCase().includes('editor')));
+  t.deepEqual(openEditorWindow.lastCall.args, [
+    incomplete,
+    {recordingName: 'Incomplete'}
+  ]);
+
+  t.true(await checkPromise);
+
+  t.false(recordingHistory.has('activeRecording'));
+  t.true(fakeService.cleanUp.calledOnceWith({some: 'state'}));
+  t.deepEqual(recordingHistory.get('recordings'), [
+    {filePath: incomplete, name: 'Incomplete', date: new Date().toISOString()}
+  ]);
+});
+
+test('checkForActiveRecording with known corrupt recording', async t => {
+  recordingHistory.set('activeRecording', {
+    filePath: corrupt,
+    name: 'Corrupt',
+    date: new Date().toISOString(),
+    apertureOptions: {},
+    plugins: {}
+  });
+
+  const checkPromise = checkForActiveRecording();
+
+  const dialogState = await dialog.waitForDialog();
+  t.true(dialogState.detail.includes('corrupt'));
+  t.true(dialogState.detail.includes('moov atom not found'));
+  t.truthy(dialogState.message);
+
+  // Don't delete it until user is done interacting with the dialog
+  t.true(recordingHistory.has('activeRecording'));
+
+  await dialog.fakeAction(dialogState.defaultId);
+
+  const newDialogState = await dialog.getCurrentState();
+  t.true(newDialogState.message.includes('unable'));
+
+  await dialog.fakeAction(newDialogState.defaultId);
+
+  t.true(shell.showItemInFolder.calledWithExactly(corrupt));
+
+  t.true(await checkPromise);
+
+  t.false(recordingHistory.has('activeRecording'));
+  t.is(recordingHistory.get('recordings').length, 0);
+});
+
+test('checkForActiveRecording with unknown corrupt recording', async t => {
```

Comment:
```suggestion
test('`checkForActiveRecording()` with unknown corrupt recording', async t => {
```

Just to make it clear it's a function. Applies in multiple places.